scripts/migrate_assets.js

```diff
@@ -0,0 +1,229 @@
+      'attributes': UserConstants.BASIC_USER_FIELDS,
+      'where': {
+        'id': fromUser
+      }
+    }
+  };
+
+  DB.Asset.findAll(options).complete(function(err, assets) {
+    if (err) {
+      return log.error({'err': err}, 'Failed to get the assets to migrate');
+    }
+
+    async.eachSeries(assets, function(asset, callback) {
+      migrateAsset(asset.toJSON(), callback);
+    }, function() {
+      log.info({'total': assets.length, 'results': results}, 'Finished migrating assets');
+    });
+  });
+};
+
+/**
+ * Migrate an asset
+ *
+ * @param  {Asset}            asset               The asset to migrate
+ * @param  {Function}         callback            Standard callback function
+ */
+var migrateAsset = function(asset, callback) {
+  var type = asset.type;
+
+  // Callback executed when the asset has finished migrating
+  var migrateCallback = function(err, newAsset) {
+    results[type] = results[type] || {
+      'success': 0,
+      'error': 0
+    };
+
+    if (err) {
+      log.error({'asset': asset, 'err': err}, 'Failed to migrate an asset');
+      results[type].error++;
+    } else {
+      log.info({'asset': asset, 'newAsset': newAsset}, 'Successfully migrated an asset');
+      results[type].success++;
+    }
+    return callback();
+  };
+
+  if (asset.type === 'link') {
+    migrateLink(asset, migrateCallback);
+  } else if (asset.type === 'file') {
+    migrateFile(asset, migrateCallback);
+  } else if (asset.type === 'whiteboard') {
+    log.info({'asset': asset}, 'Skipping whiteboard migration');
+    return callback();
+  } else {
+    log.error({'asset': asset}, 'Unrecognized asset type');
+    return callback();
+  }
+};
+
+/**
+ * Migrate a link asset
+ *
+ * @param  {Asset}            link                The link to migrate
+ * @param  {Function}         callback            Standard callback function
+ */
+var migrateLink = function(link, callback) {
+  AssetsAPI.createLink(ctx, link.title, link.url, {
+    'description': link.description || undefined,
+    'source': link.source || undefined,
+    'thumbnail_url': link.thumbnail_url || undefined,
+    'image_url': link.image_url || undefined,
+    'embed_id': link.embed_id || undefined,
+    'embed_key': link.embed_key || undefined,
+    'embed_code': link.embed_code || undefined
+  }, callback);
+};
+
+/**
+ * Migrate a file asset
+ *
+ * @param  {Asset}            file                The file to migrate
+ * @param  {Function}         callback            Standard callback function
+ */
+var migrateFile = function(file, callback) {
+  // Download the file to a temporary folder
+  var path = temp.path();
+  request(file.download_url).pipe(fs.createWriteStream(path))
+    .on('error', callback)
+    .on('finish', function() {
+
+      // Create the file asset
+      AssetsAPI.createFile(ctx, file.title, {
+        'mimetype': file.mime,
+        'file': path,
+        'filename': file.title
+      }, {
+        'description': file.description || undefined,
```

Comment:
We're calling the internal API directly and Joi throws an error for optional values when `null` is provided. I didn't really want to add `allow(null)` to all validation rules, so I just default them to `undefined` for all falsey values. However, I'm happy to go for an alternative solution here. What do you think?